include/libtorrent/utp_socket_manager.hpp

```diff
@@ -130,23 +130,35 @@ namespace libtorrent
 		typedef std::multimap<std::uint16_t, utp_socket_impl*> socket_map_t;
 		socket_map_t m_utp_sockets;
 
+		using socket_vector_t = std::vector<utp_socket_impl*>;
+
 		// this is a list of sockets that needs to send an ack.
 		// once the UDP socket is drained, all of these will
 		// have a chance to do that. This is to avoid sending
 		// an ack for every single packet
-		std::vector<utp_socket_impl*> m_deferred_acks;
+		socket_vector_t m_deferred_acks;
+
+		// storage used for saving cpu time on "push_back"
+		// by using already pre-allocated vectors
+		socket_vector_t m_socket_drained_temp;
```

Comment:
I was going to suggest going back to a member here. This should fix the problem, since each utp_socket_manager has its own temporary storage.